src/extension/openFileAtLocation.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {RemoteExtension} from "../common/remoteExtension";
+import {ReactNativeProjectHelper} from "../common/reactNativeProjectHelper";
+import {InternalErrorCode} from "../common/error/internalErrorCode";
+import {ErrorHelper} from "../common/error/errorHelper";
+import * as path from "path";
+
+/* Usage:
+...path\openFileAtLocation.js filename:lineNumber
+...path\openFileAtLocation.js filename
+*/
+
+if (process.argv.length < 3) {
+    throw "Wrong number of parameters provided. Please refer to the usage of this script for proper use.";
+}
+const fileInfo = process.argv[2].split(":");
+const filename: string = fileInfo[0];
+let lineNumber: number = 1;
+
+if (fileInfo.length >= 2) {
+    lineNumber = parseInt(fileInfo[1], 10);
+}
+
+getReactNativeWorkspaceForFile(filename).then(projectRootPath => {
+    const remoteExtensionServerPath = RemoteExtension.atProjectRootPath(projectRootPath);
+    return remoteExtensionServerPath.openFileAtLocation(filename, lineNumber);
+}).done(() => { }, (reason) => {
+    throw ErrorHelper.getNestedError(reason, InternalErrorCode.CommandFailed, "Unable to communicate with VSCode. Please make sure it is open in the appropriate workspace.");
+});
+
+function getReactNativeWorkspaceForFile(file: string): Q.Promise<string> {
+    return getNameForRNParentWorkspace(path.dirname(file)).then(workspaceName => {
+        if (workspaceName) {
+            return workspaceName;
+        }
+        return Q.reject<string>("Could not find workspace for file " + file);
+    });
+}
+
+function getNameForRNParentWorkspace(dir: string): Q.Promise<string> {
+    const reactNativeProjectHelper = new ReactNativeProjectHelper(dir);
+    return reactNativeProjectHelper.isReactNativeProject().then(isRNProject => {
+        if (isRNProject) {
+            return dir;
+        }
+        if (dir === "" || dir === "." || dir === "/" || dir === path.dirname(dir)) {
+            return null;
```

Comment:
By doing this I would loose the `file` in the reject message (because of the recursive nature of this method)